server/chat-plugins/info.js

```diff
@@ -2380,6 +2386,43 @@ const commands = {
 	],
 };
 
+/** @type {PageTable} */
+const pages = {
+	punishments(query, user, connection) {
+		let buf = "";
+		this.extractRoom();
+		if (!user.named) return Rooms.RETRY_AFTER_LOGIN;
+		buf += `|title|Punishments\n|pagehtml|<div class="pad"><h2>List of active punishments:</h2>`;
+		if (!user.can('mute')) {
+			return buf + `<div class="notice message-error">Access denied.</div>`;
+		}
+		const subMap = Punishments.roomIps.get(this.room.id);
+		const muteQueue = this.room.muteQueue;
+		if (subMap) {
+			for (const punishment of subMap) {
```

Comment:
this leads to duplicate punishments. What you should be doing is first iterate over all of them, and make a temporary storage of all the (unique) punishments and the names/ips associated with them, then display those. Otherwise bans on a school ip are going to show up once on the list per name caught in the ban.